### ISSUE TITLE
User-defined PATH no longer sourced in remote ssh session after 1.87.1 release

### ISSUE THREAD

**Issue by JtMotoX** [microsoft/vscode#207117]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.87.1
- OS Version (local): Windows 11
- OS Version (remote ssh): AlmaLinux 8.9

Steps to Reproduce:

Server Steps (Linux):
1. Make sure git is not installed
1. Install git in a non-standard path (git installed via nix in my case)
1. Update your profile file (.zshrc, .bashrc, .bash_profile, etc..) and update your PATH to include the non-standard directory where you installed git.
1. Clone any git repo

Client Steps (Windows):
1. Install code 1.87.1
1. Remote SSH to the linux server
1. Open the git repo you cloned earlier
1. VSC will not detect that it is a git repo and complains that git is not installed.

I have not run into this issue until updating to 1.87.1

![image](https://github.com/microsoft/vscode/assets/7191259/b24af8c5-359a-4de1-b941-88ff535c517d)

**Comment by JtMotoX** [comment]
![image](https://github.com/microsoft/vscode/assets/7191259/773e7547-513e-4a5a-ab25-6dd0a99bbdbc)

**Comment by JtMotoX** [comment]
I finally found the issue. My custom systemd service script was executing the `code serve-web` command in a non-interactive shell (without login).  I added the `-l` flag for a non-interactive login shell.

Before:
```bash
COMMAND="${CODE_CLI} serve-web . . ."
${COMMAND}
```

After:
```bash
COMMAND="${CODE_CLI} serve-web . . ."
${SHELL} -l -c "${COMMAND}"
```

Now that the process is "logging in" and loading my zsh profile, my `~/nix-profile/bin` directory is now included in my PATH. Now git and git-crypt are working as expected.
